### PR TITLE
Feature/17 5 implement method for moderating existing time slot

### PR DESCRIPTION
**Pull request by eugent482**

### CODE REVIEW

**Review by TREN-D — APPROVED** [review]

**Review by SergiyKozachok — APPROVED** [review]

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 1]
```diff
@@ -14,6 +14,9 @@ public class AppointmentService : IAppointmentService
         private const int MINIMUM_APPOINTMENT_TIME = 15;
         private const int DEFAULT_STATUS = 1;
         private const int RESERVED_STATUS = 2;
+        private const int CANCELED_STATUS = 3;
+        private const int FINISHED_STATUS = 4;
```
Comment (OleksandrAntoniuk) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs:20]: really bad approach. Why we can't create enum with all status results?

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 2]
```diff
@@ -97,6 +100,99 @@ public IQueryable<AvailableAppointmentModel> GetAvailableAppointments(int doctor
...
+        /// <param name="appointmentId"></param>
+        /// <param name="doctorId"></param>
+        /// <returns></returns>
+        public (bool status, string message) DeleteAppointment(int appointmentId, int doctorId)
+        {
+            var appointment = _context.Appointments
+                .FirstOrDefault(a => a.Id == appointmentId);
+            if (appointment == null)
+            {
+                return (false, "Appointment not found");
```
Comment (OleksandrAntoniuk) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs:130]: Back-end web api shouldn't return string messages visible to user. It's front-end job

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 3]
```diff
@@ -97,6 +100,99 @@ public IQueryable<AvailableAppointmentModel> GetAvailableAppointments(int doctor
             return appointments;
         }
 
+        /// <summary>
+        /// Delete doctor's appointment if it's not reserved
+        /// </summary>
+        /// <param name="appointmentId"></param>
+        /// <param name="doctorId"></param>
+        /// <returns></returns>
+        public (bool status, string message) DeleteAppointment(int appointmentId, int doctorId)
```
Comment (OleksandrAntoniuk) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs:124]: Change signature to return only status result

**Review by serhiidotsyk — APPROVED** [review]

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 4]
```diff
@@ -97,6 +100,99 @@ public IQueryable<AvailableAppointmentModel> GetAvailableAppointments(int doctor
             return appointments;
         }
 
+        /// <summary>
+        /// Delete doctor's appointment if it's not reserved
+        /// </summary>
+        /// <param name="appointmentId"></param>
+        /// <param name="doctorId"></param>
+        /// <returns></returns>
```
Comment (OleksandrAntoniuk) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs:123]: why return field is empty?

File: InternetHospital.WebApi/Controllers/AppointmentsController.cs [hunk 5]
```diff
@@ -62,5 +62,74 @@ public IActionResult AddAppointment([FromBody] AppointmentCreationModel creation
...
+                }
+            }
+            else
+            {
+                return BadRequest(new { message = "Wrong claims" });
+            }
+        }
+
+        [Authorize(Policy = "ApprovedDoctors")]
+        [HttpPost("delete")]
```
Comment (OleksandrAntoniuk) [inline InternetHospital.WebApi/Controllers/AppointmentsController.cs:90]: why http post?

File: InternetHospital.BusinessLogic/Services/AppointmentService.cs [hunk 6]
```diff
@@ -51,6 +57,8 @@ public IQueryable<AppointmentModel> GetMyAppointments(int doctorId)
         /// <returns></returns>
         public (bool status, string message) AddAppointment(AppointmentCreationModel creationModel, int doctorId)
         {
+            DeleteUncommittedAppointments(doctorId);
```
Comment (OleksandrAntoniuk) [inline InternetHospital.BusinessLogic/Services/AppointmentService.cs:60]: what if this method crashes?

File: InternetHospital.WebApi/Controllers/AppointmentsController.cs [hunk 7]
```diff
@@ -62,5 +62,74 @@ public IActionResult AddAppointment([FromBody] AppointmentCreationModel creation
...
+
+        [Authorize(Policy = "ApprovedDoctors")]
+        [HttpPost("finish")]
+        public IActionResult FinishAppointment([FromBody] AppointmentEditingModel model)
+        {
+            if (int.TryParse(User.Identity.Name, out var doctorId))
+            {
+                var (status, message) = _appointmentService.FinishAppointment(model.Id, doctorId);
+
+                if (status)
```
Comment (OleksandrAntoniuk) [inline InternetHospital.WebApi/Controllers/AppointmentsController.cs:120]: too much if statements. Please reduce complexity

**Review by OleksandrAntoniuk — COMMENTED** [review]
do you see my coments?

**Pull request merged**